src/app/feed/shared/feed-connection.model.spec.ts

```diff
@@ -30,102 +30,248 @@ describe("Service: FeedConnection", () => {
-    let fConnection: FeedConnection = new FeedConnection();
-    fConnection.setAttrs(
-      this.pluginHandle,
-      1
-    );
+  describe("#publish", () => {
+    beforeEach(() => {
+      this.options = jasmine.createSpyObj("options", ["success", "error"]);
+    });
 
-    let jsep: any = { id: 13 };
-    fConnection.handleRemoteJsep(jsep);
+    it("should create a webRTC offer as subscriber", () => {
+      this.connection.publish();
+
+      expect(this.pluginHandle.createOffer).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          media: {
+            videoRecv: false,
+            audioRecv: false,
+            video: "screen",
+            audioSend: false,
+            data: false
+          },
+          success: jasmine.any(Function),
+          error: jasmine.any(Function)
+        })
+      );
+    });
+
+    it("should create a webRTC offer as main", () => {
+      this.connection.role = "main";
+      this.connection.publish();
+
+      expect(this.pluginHandle.createOffer).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          media: {
+            videoRecv: false,
+            audioRecv: false,
+            audioSend: true,
+            videoSend: true,
+            data: true
+          },
+          success: jasmine.any(Function),
+          error: jasmine.any(Function)
+        })
+      );
+    });
+
+    it("should call success callback when webRTC offer success", () => {
+      this.pluginHandle.createOffer.and.callFake((options) => {
+        options.success()
+      });
+      this.connection.publish(this.options);
+
+      expect(this.options.success).toHaveBeenCalled();
+    });
+
+    it("should call success callback when webRTC offer success", () => {
+      this.pluginHandle.createOffer.and.callFake((options) => {
+        options.error()
+      });
+      this.connection.publish(this.options);
+
+      expect(this.options.error).toHaveBeenCalled();
+    });
 
-    expect(this.pluginHandle.handleRemoteJsep).toHaveBeenCalledWith(
-      jasmine.objectContaining({
-        jsep: jsep
-      })
-    );
   });
 
-  it("should call pluginHandle.data when call sendData", () => {
-    let fConnection: FeedConnection = new FeedConnection();
-    fConnection.setAttrs(
-      this.pluginHandle,
-      1
-    );
+  describe("#subscribe", () => {
+    beforeEach(() => {
+      this.jsepVal = {};
+    });
+    it("should create a webRTC answer", () => {
+      this.connection.subscribe(this.jsepVal);
+
+      expect(this.pluginHandle.createAnswer).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          jsep: this.jsepVal,
+          media: {
+            audioSend: false,
+            videoSend: false,
+            data: true
+          },
+          success: jasmine.any(Function),
+          error: jasmine.any(Function)
+        })
+      );
+    });
+
+    it("should send a request start message when webRTC answer success", () => {
```

Comment:
As above, I think this test does not cover the original code, right? Testing anonymous callbacks in this case (depending on `pluginHandle` and connections) can be tricky. I think that we should try another approach (de-anonymizing the callbacks, for example).

However, I don't want this issue to block us, so I propose postponing these tests (remove the current ones). Please, add a Trello card to not forget about them.